wallet/cardano-wallet.js

```diff
@@ -189,7 +196,25 @@ const CardanoWallet = async (options) => {
     */
     const deviation = 4
 
-    return txFeeFunction(txSizeInBytes + deviation)
+    let fee = txFeeFunction(txSizeInBytes + deviation)
+
+    /*
+    * if they are equal it means there is no change necessary
+    * if they are more the transaction is invalid even with higher fee so we let caller handle it
+    */
+    if (coins + fee >= txInputsCoinsSum) {
+      return fee
+    } else {
+      fee = txFeeFunction(txSizeInBytes + txOutputsSize + deviation) //we try to compute fee for 2 output tx
+      if (coins + fee > txInputsCoinsSum) {
```

Comment:
this case should also be handled when preparing the transaction outputs. Currently if coins + fee < tx inputs sum, the final transaction contains two outputs. Maybe this function could output an object instead of a single number to provide additional information to the caller, something like this: 
{fee: 155547, outputsCount: 2, valid: false}

Reply:
i think that bloats the scope of this issue a bit, id say make it a separate issue depending on this.

Reply:
ok, I misunderstood the code